ide-common/src/main/kotlin/org/digma/intellij/plugin/errorreporting/ErrorReporter.kt

```diff
@@ -0,0 +1,98 @@
+package org.digma.intellij.plugin.errorreporting
+
+import com.github.benmanes.caffeine.cache.Caffeine
+import com.intellij.ide.impl.ProjectUtil
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.service
+import com.intellij.openapi.diagnostic.UntraceableException
+import com.intellij.openapi.project.Project
+import org.digma.intellij.plugin.posthog.ActivityMonitor
+import java.lang.reflect.Field
+import java.util.concurrent.TimeUnit
+import java.util.concurrent.atomic.AtomicInteger
+
+@Service(Service.Level.APP)
+class ErrorReporter {
+
+
+    companion object {
+
+        @JvmStatic
+        fun getInstance(): ErrorReporter {
+            return service<ErrorReporter>()
+        }
+    }
+
+
+    fun reportError(project: Project?, message: String, t: Throwable) {
+        if (isTooFrequentException(message, t)) {
+            return
+        }
+
+        //todo: change ActivityMonitor to application service
+
+        val projectToUse = project ?: ProjectUtil.getActiveProject()
```

Comment:
`ProjectUtil.getActiveProject()`
that's a great function!
no need to pass it every time

Reply:
need , because we don't know what ProjectUtil.getActiveProject will return, its used only if there is no project reference when calling reoportError.
I think it will be removed when ActivityMonitor is changed to application service